crates/vision/src/image_segmenter.rs

```diff
@@ -142,6 +203,31 @@ fn new_grid(
     }
 }
 
+fn next_y_from_current_y(
```

Comment:
maybe indicate in the name this is only for horizontal segments. Something like `ext_horizontal_segment_height`